packages/api/shared/access-control/authorize.ts

```diff
@@ -0,0 +1,77 @@
+import { subject as buildSubject } from '@casl/ability';
+import type { RouteRequest } from '../Route';
+import { client } from '../db';
+import { ForbiddenError, NotFoundError } from '../errors';
+import { Action, createPolicyFor, RawSubject, Subject } from './policy';
+
+interface AuthorizeOptions {
+  action: Action;
+  subject: RawSubject;
+  subjectId?: string;
+}
+
+/**
+ * Determine whether a user is authorized to perform the requested action on the given subject.
+ *
+ * If no subject ID is given, the user will be authorized if they have access to _any_ (not all) of that subject type.
+ * @param options
+ */
+export function authorize<Params, Body>(
+  options:
+    | AuthorizeOptions
+    | ((req: RouteRequest<Params, Body>) => AuthorizeOptions)
+): (req: RouteRequest<Params, Body>) => Promise<void> {
```

Comment:
This authorize function makes authorization easy for routes. You just need to convert the request into an object with the action, and subject ID. This function will load the correct data to send through the policy and throw Forbidden and NotFound errors as appropriate